va/va_trace.c

```diff
@@ -44,16 +44,34 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
+#if defined(_WIN32)
+#include <windows.h>
+#include <io.h>
+#include <time.h>
+typedef CRITICAL_SECTION mutex_t;
+inline void mutex_lock(mutex_t* a) { EnterCriticalSection(a); }
+inline void mutex_unlock(mutex_t* a) { LeaveCriticalSection(a); }
+inline void mutex_init(mutex_t* a) { InitializeCriticalSection(a); }
+inline void mutex_destroy(mutex_t* a) { DeleteCriticalSection(a); }
```

Comment:
Similar to the `compat_win32.h` example above - these are better moved out, where a `pthread_foo` interface is provided or emulated.

The train of thought is that the onus of lacking standard API is on the platform, not on everyone else.